### PR TITLE
Add org.adishatz.Screenshot

### PR DESCRIPTION
**Pull request by bellegarde-c**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly.
 Screenshot application:
* Ability to save to path
* Ability to copy to clipboard on Phosh

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: org.adishatz.Screenshot.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        },
+        {
+            "name" : "screenshot",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://gitlab.gnome.org/gnumdk/screenshot.git",
+		    "tag": "0.1"
```
Comment (bbhtt) [inline org.adishatz.Screenshot.json:51]: Indentation is broken here.

File: org.adishatz.Screenshot.json [hunk 2]
```diff
@@ -0,0 +1,56 @@
+{
+    "app-id" : "org.adishatz.Screenshot",
```
Comment (bbhtt) [inline org.adishatz.Screenshot.json:2]: How do I know you own adishatz.org ? The webpage there is 403 https://adishatz.org/

Please put something up there like a link to https://gitlab.gnome.org/gnumdk/screenshot.git

**Comment by bbhtt** [comment]
bot, build org.adishatz.Screenshot

File: org.adishatz.Screenshot.json (hunk 2 above)
Comment (bellegarde-c) [reply org.adishatz.Screenshot.json:2]: I already have an application with my domain at https://flathub.org/apps/org.adishatz.syncpasswd

**Comment by bellegarde-c** [comment]
> bot, build org.adishatz.Screenshot

I have fixed the missing fallback-x11 issue.

For xdg-cache, I use it to ask "Screenshot portal" to save file here, so my app is able to read/move it.

If you have any other better suggestion, I will update the code.

File: org.adishatz.Screenshot.json (hunk 2 above)
Comment (bbhtt) [reply org.adishatz.Screenshot.json:2]: That's not verified so it doesn't  prove ownership

**Comment by bbhtt** [comment]
> For xdg-cache, I use it to ask "Screenshot portal" to save file here, so my app is able to read/move it.

Where in xdg-cache? It'd better if it is in a subfolder there for your app.

Also why not ask it to save in `~/.var/app/app_id/cache`?

**Comment by bellegarde-c** [comment]
> > For xdg-cache, I use it to ask "Screenshot portal" to save file here, so my app is able to read/move it.
> 
> Where in xdg-cache? It'd better if it is in a subfolder there for your app.
> 
> Also why not ask it to save in `~/.var/app/app_id/cache`?

Indeed, did not notice that g_get_user_cache_dir () does not return ~/.cache on flatpak. So option was not needed.

File: org.adishatz.Screenshot.json (hunk 2 above)
Comment (bbhtt) [reply org.adishatz.Screenshot.json:2]: Thanks!

**Comment by bbhtt** [comment]
bot, build org.adishatz.Screenshot

**Comment by bbhtt** [comment]
Lots of appstream related errors:

```json
{
    "errors": [
        "appstream-id-mismatch-flatpak-id",
        "appstream-missing-categories",
        "appstream-failed-validation"
    ],
    "warnings": [
        "appstream-screenshot-missing-caption",
        "desktop-file-low-quality-category"
    ],
    "appstream": [
        "E: org.adishatz.Screenshot.desktop:65: release-description-outside-tag p",
        "E: org.adishatz.Screenshot.desktop:19: screenshot-image-no-source",
        "E: org.adishatz.Screenshot.desktop:~: desktop-app-launchable-omitted",
        "W: org.adishatz.Screenshot.desktop:18: screenshot-default-missing",
        "E: org.adishatz.Screenshot.desktop:62: custom-key-duplicated Purism::form_factor",
        "E: org.adishatz.Screenshot.desktop:20: screenshot-image-invalid-type default"
    ],
    "info": [
        "appstream-missing-categories: The catalogue file is missing categories. Please check if categories are defined in the desktop file or the metainfo file and it does not have, only low quality categories",
        "appstream-failed-validation: Metainfo file /tmp/tmpmpojv9qm/metainfo/org.adishatz.Screenshot.metainfo.xml has failed validation. Please see the errors in appstream block",
        "appstream-screenshot-missing-caption: One or more screenshots are missing captions in the Metainfo file",
        "appstream-id-mismatch-flatpak-id: The value of ID tag: org.adishatz.Screenshot.desktop in Metainfo does not match the FLATPAK_ID: org.adishatz.Screenshot. Please see the docs for more details",
        "desktop-file-low-quality-category: A low quality category was found in the desktop file: {'GTK'}"
    ],
    "message": "Please consult the documentation at https://docs.flathub.org/docs/for-app-authors/linter"
}
```

desktop-file-low-quality category and category missing are opposites sides of same coin. If you add some non low quality categories in the desktop file, both of them will be gone. See https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#categories-and-keywords

**Comment by bellegarde-c** [comment]
bot, build org.adishatz.Screenshot

**Pull request closed without merging**